### PR TITLE
Rework Nuttx config

### PR DESCRIPTION
**Pull request by LaszloLango**
Nuttx build should use it's own JerryScript port instead of
the default port from JerryScript. Nuttx build should not use
jerry-libc. Use gcc only for Nuttx builds.

IoT.js-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
@galpeter or @akosthekiss could you take a look on the cmake changes?

**Comment by LaszloLango** [comment]
| File        |   Old size | New size |
|-|-|-|
|nuttx.bin |    415 204 |    412 396 |
|nuttx.hex | 1 167 937 |  1 160 033 |

File: cmake/jerry.cmake [hunk 1]
```diff
@@ -83,6 +83,13 @@ if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
   list(APPEND DEPS_LIB_JERRY_ARGS -DFEATURE_ERROR_MESSAGES=ON)
 endif()
 
+# Nuttx is not using the default port implementation of JerryScript
+if("${TARGET_OS}" MATCHES "NUTTX")
+  list(APPEND DEPS_LIB_JERRY_ARGS -DJERRY_PORT_DEFAULT=OFF)
```
Comment (galpeter) [inline cmake/jerry.cmake]: I'm not sure, but this shouldn't this also apply to TizenRT?

Comment (LaszloLango) [inline cmake/jerry.cmake]: Good question. Unfortunately I'm not familiar with TizenRT. CC: @yichoi, @glistening

**Comment by LaszloLango** [comment]
Note: #848 must land first.

File: config/nuttx/stm32f4dis/app/jerry_port.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  exit(1);
+} /* jerry_port_fatal */
+
+/**
+ * Provide log message implementation for the engine.
+ */
+void jerry_port_log(jerry_log_level_t level, /**< log level */
+                    const char *format,      /**< format string */
+                    ...)                     /**< parameters */
+{
```
Comment (zherczeg) [inline config/nuttx/stm32f4dis/app/jerry_port.c]: the `{` is after the `)` in the other lines.

File: config/nuttx/stm32f4dis/app/jerry_port.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  /* drain log messages since IoT.js does not support log levels */
```
Comment (zherczeg) [inline config/nuttx/stm32f4dis/app/jerry_port.c]: has not ... yet

File: config/nuttx/stm32f4dis/app/iotjs_main.c [hunk 4]
```diff
@@ -50,31 +50,30 @@
...
+#include <nuttx/config.h>
 
 #include <stdio.h>
 #include "setjmp.h"
 
 #ifdef CONFIG_IOTJS
-# if !defined(CONFIG_HAVE_CXX) || !defined(CONFIG_HAVE_CXXINITIALIZE)
-#   error Need CONFIG_HAVE_CXX and CONFIG_HAVE_CXXINITIALIZE
-# endif
+#if !defined(CONFIG_HAVE_CXX) || !defined(CONFIG_HAVE_CXXINITIALIZE)
```
Comment (zherczeg) [inline config/nuttx/stm32f4dis/app/iotjs_main.c]: Do we still need this define check?

**Comment by yichoi** [comment]
@LaszloLango rebase please

File: config/nuttx/stm32f4dis/app/iotjs_main.c (hunk 4 above)
Comment (yichoi) [inline config/nuttx/stm32f4dis/app/iotjs_main.c]: No. I forgot to remove it. please remove it

**Comment by LaszloLango** [comment]
Update: The latest changes haven't been tested on device yet.

File: cmake/jerry.cmake (hunk 1 above)
Comment (glistening) [inline cmake/jerry.cmake]: @LaszloLango I will check it with tizenrt-artik05x.

Comment (glistening) [inline cmake/jerry.cmake]: @LaszloLango I am getting several errors during build for tizenrt-artik05x. We will make another PR for tizen-artik05x. Let land this PR separately.

Comment (LaszloLango) [inline cmake/jerry.cmake]: @glistening ok, thanks for checking it.

**Comment by LaszloLango** [comment]
The patch works well on device.

**Comment by yichoi** [comment]
LGTM
